### DISCUSSION TITLE
Upgrading from 1.6.2 to 2.0.0. Is it worth?

### DISCUSSION THREAD

**Discussion by tsoodla** [bellrichm/WeeWX-MQTTSubscribe#134] [Q&A] [answered]
I have no issues whatsoever running MQTTSubscribe as the driver but a new version is out with a lot of configuration changes and config depreciations. So the question is - is it worth to upgrade from 1.6.2 to 2.0.0?

**Comment by bellrichm** [comment, marked as answer]
If everything is working, you don’t have to upgrade. Just remember that any fixes will only be applied to 2.x. So, you might want to upgrade when it is convenient/you have time.

While the 1.6.2 weewx.conf may not work with 2.x, the 2.x weewx.conf should be fully compatible with 1.6.2. So, you should be able to update your weewx.conf and then upgrade to 2.x.

Also at startup of 1.6.2, any options that are not valid in 2.x should be logged.

Lastly, if you post you configuration (removing any sensitive information) I’d be happy to help ensure it is 2.x compatible.

rich

**Reply by tsoodla** [reply]
The current config is as follows:


```
[MQTTSubscribeDriver]
    # This section is for the MQTTSubscribe driver.
    
    # The driver to use:
    driver = user.MQTTSubscribe
    
    # The MQTT server.
    # Default is localhost.
    host = localhost
    
    # The port to connect to.
    # Default is 1883.
    port = 1883
    
    # Maximum period in seconds allowed between communications with the broker.
    # Default is 60.
    keepalive = 60
    
    # username for broker authentication.
    # Default is None.
    username = xxxxx
    
    # password for broker authentication.
    # Default is None.
    password = xxxxx
    
    # Configuration for the message callback.
    [[message_callback]]
        type = individual
    
    # The topics to subscribe to.
    [[topics]]
        unit_system = METRIC
        use_topic_as_fieldname = true
        use_server_datetime = true
        
        [[[rtl_433/lab/devices/Solight-TE44/1/4/temperature_C]]]
            name = outTemp
        [[[emhi/26038/uvindex]]]
            name = UV
        [[[emhi/26038/airpressure]]]
            name = pressure
        [[[emhi/26038/precipitations]]]
            name = rain
            units = mm
            contains_total = true
        [[[emhi/26038/relativehumidity]]]
            name = outHumidity
        [[[emhi/26038/winddirection]]]
            name = windDir
        [[[emhi/26038/windspeed]]]
            name = windSpeed
            units = meter_per_second
        [[[emhi/26038/windspeedmax]]]
            name = windGust
            units = meter_per_second
        [[[emhi/26038/phenomenon]]]
            name = phenomenon
            conversion_type = None
        [[[emhi/26038/visibility]]]
            name = visibility
            conversion_type = None

[Accumulator]
    [[phenomenon]]
        accumulator = firstlast
        extractor = last
    [[visibility]]
        accumulator = firstlast
        extractor = last

```
Thank you!

**Reply by bellrichm** [reply]
I took your configuration and ran it through 1.6.2 and got no warnings that any options were to be removed. I also ran it through 2.x and did not get any errors.
So, hopefully you **should not** have any problems upgrading...
rich

**Reply by tsoodla** [reply]
Great. It works. I just removed  `use_topic_as_fieldname = true`
Thank you!

**Reply by bellrichm** [reply]
Thanks for letting me know. Yeah, in 2.x `use_topic_as_fieldname` is no longer necessary. Since it doesn't hurt having it, I did not bring it up.
